packages/purpose-process/src/services/purposeService.ts

```diff
@@ -252,9 +245,8 @@ export function purposeServiceBuilder(
       );
 
       const activeProducerDelegation =
-        await readModelService.getActiveDelegation(
-          purpose.data.eserviceId,
-          delegationKind.delegatedProducer
+        await readModelService.getActiveProducerDelegationByEserviceId(
```

Comment:
Waiting to settle a discussion we are having offline -- probably, also in this call `getPurposeById` we should reuse `assertRequesterIsAllowedToRetrieveRiskAnalysisDocument` and somehow refactor/remove `authorizeRiskAnalysisForm` because also here the consumer delegator shall probably be allowed to retrieve the purpose

Reply:
Same for some other calls, for example `deletePurposeVersion`, where the check is still only on the consumer and it does not use the new validator... I would double-check all the calls in this file and make sure they use the new validators